keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1210,230 +955,202 @@ server-side-apply (SSA) simpler. Patching individual list elements is not
+// a pool name is chosen by the driver as necessary. It can be left blank if the
+// device names are unique in the entire cluster. The unique identifier then
+// has the same format as before with `<driver name>[/<pool name>]/<device name>`.
 type ResourceSlice struct {
     metav1.TypeMeta
     // Standard object metadata
+    // +optional
     metav1.ObjectMeta
 
-    // NodeName identifies the node which provides the resources
-    // if they are local to a node.
-    //
-    // A field selector can be used to list only ResourceSlice
-    // objects with a certain node name.
-    NodeName string
+    Spec ResourceSliceSpec
+
+    // A status might get added later.
+}
 
+type ResourceSliceSpec struct {
     // DriverName identifies the DRA driver providing the capacity information.
     // A field selector can be used to list only ResourceSlice
     // objects with a certain driver name.
+    //
+    // Must be a DNS subdomain and should end with a DNS domain owned by the
+    // vendor of the driver.
     DriverName string
 
-    ResourceModel
+    // PoolName is used to identify devices. For node-local devices, this
+    // is often the node name, but this is not required.
+    //
+    // It must not be longer than 253 and must consist of one or more DNS sub-domains
+    // separated by slashes.
+    PoolName string
+
+    // NodeName identifies the node which provides the devices.
+    // A field selector can be used to list only ResourceSlice
+    // objects belonging to a certain node.
+    //
+    // This field can be used to limit access from nodes to slices with
+    // the same node name. It also indicates to autoscalers that adding
+    // new nodes of the same type as some old node might also make new
+    // devices available.
+    //
+    // NodeName and NodeSelector are mutually exclusive.
+    // If both are unset, then the devices are available in the
+    // entire cluster.
+    //
+    // +optional
+    NodeName string
+
+    // Defines which nodes have access to the devices in the pool.
+    // Must not be set when the node name is set.
+    //
+    // NodeName and NodeSelector are mutually exclusive.
+    // If both are unset, then the devices are available in the
+    // entire cluster.
+    //
+    // +optional
+    NodeSelector *v1.NodeSelector
+
+    // The generation gets bumped in all slices of a pool whenever device
+    // definitions change. A consumer must only use device definitions from slices
+    // with the highest generation number and ignore all others.
+    PoolGeneration int64
+
+    // The total number of devices in the pool.
+    // Consumers can use this to check whether they have
+    // seen all slices.
+    PoolDeviceCount int64
+
+    // Devices lists all available devices in this pool.
+    //
+    // Must not have more than 128 entries.
+    Devices []Device
+
+    // FUTURE EXTENSION: some other kind of list, should we ever need it.
```

Comment:
Let's include the [partitionable device model](https://github.com/kubernetes-sigs/wg-device-management/pull/27/files#r1634768662) here, too.

Reply:
I can add here that we want to add it, but not the model itself - I don't feel qualified to explain it.

Reply:
I will look at adding text for it tomorrow.

Reply:
I'm quite confused by https://github.com/kubernetes-sigs/wg-device-management/pull/27/files#r1634768662 as it stands now and left several comments there.

Reply:
I am working on the text here:
https://github.com/pohly/enhancements/pull/18

Reply:
Thanks! I included it.